crates/wasmtime/src/compile.rs

```diff
@@ -330,13 +335,20 @@ impl<'a> CompileInputs<'a> {
 
     /// Create the `CompileInputs` for a core Wasm module.
     fn for_module(
+        triple: &target_lexicon::Triple,
         types: &'a ModuleTypesBuilder,
         translation: &'a ModuleTranslation<'a>,
         functions: PrimaryMap<DefinedFuncIndex, FunctionBodyData<'a>>,
     ) -> Self {
-        let mut ret = Self::default();
-        let module_index = StaticModuleIndex::from_u32(0);
+        let mut ret = CompileInputs {
+            need_wasm_to_array_trampolines: !matches!(
+                triple.architecture,
+                target_lexicon::Architecture::Pulley32 | target_lexicon::Architecture::Pulley64
```

Comment:
Do you think it's worth having for example an extension trait on `TargetTriple` for `.is_pulley()` to use here and in wasmtime-environ as well? Not that this set of architectures is going to change over time really

Reply:
Good idea, will do in a follow up